### DISCUSSION TITLE
About web installer

### DISCUSSION THREAD

**Discussion by spanjaan** [wintercms/winter#251] [General]
![Screenshot (19)](https://user-images.githubusercontent.com/64153858/126668957-12fd6de9-4a3d-4bc7-b59a-3b9cd15ef47e.png)
when in install my hosting account i have get this error.  how to fix this problem?

thank you.

**Comment by LukeTowers** [comment]
This should be an issue in wintercms/web-installer. @bennothommo any input?

**Comment by bennothommo** [comment]
@Anjaan2077 would you mind re-running the Web Installer with the Network tab in your browser's Inspector Tools open, and paste in the response you get for the AJAX command that fails with that error?

Likely scenario here is that a PHP error message is getting printed out in the JSON response.

**Comment by gusdgg** [comment]
I've got the same error.  Checking the network tab as suggested by @bennothommo I found a post to api.php with status 500 and no response. So, I've checked the apache error log and I've found this:

PHP Fatal error:  Maximum execution time of 120 seconds exceeded in C:\\xampp74\\htdocs\\winter\\install\\api\\src\\Api.php on line 225, referer: http://localhost/winter/install.html

It seems to be a timeout problem when downloading the files from github. So, increasing the max_execution_time in php.ini solve my problem.

**Reply by RomainMazB** [reply]
@gusdgg Thanks for this information, I've made a PR to the documentation to add this possible cause.

https://github.com/wintercms/docs/pull/21

**Reply by bennothommo** [reply]
@gusdgg just for testing purposes, would you be able to set up a PHP script like so, and just let me know how long it takes to run?

```
<?php
file_put_contents(
    __DIR__ . '/winter.zip',
    file_get_contents('https://github.com/wintercms/winter/archive/refs/heads/1.1.zip')
);
```

It'll write a `winter.zip` file to the current directory where the PHP file is stored. The reason for this test is that a) I want to make sure it's not another issue (such as `allow_url_fopen` being disabled) and b) I find it rather suprising that it would take so long to download a ~9MB file that it reaches the max execution time.

**Reply by igorsamardzija** [reply]
@gusdgg & @RomainMazB 
I set Maximum execution time to 120 and getting same error, so this wasn`t solution for me. Any other workaround?

**Reply by gusdgg** [reply]
@igorsamardzija I think that 120 seconds its the default value, so try again increasing that value to 360 or similar.

**Reply by gusdgg** [reply]
@bennothommo I set up the php script that you suggested and today It takes exactly 118s. I think just like you, It's rare that a 9MB file takes so long. For the record, I've made various installations without any problems before this one. The day the insallation failed I've increased the max execution time to 180 allowing the installer to download the winter.zip without errors but  then failed again when the installer tried to download composer required packages. Increasing the max execution time again, but this time to 360, the installation finished without errors. So, I think probably it's something related with the internet connection (this was a test server at home, not a host provider) or slow traffic between my internet provider and github.com

**Reply by bennothommo** [reply]
@gusdgg Thanks heaps for the info - I'll push out a fix for now with an increased time limit on all API calls to see if it resolves most reported issues.

**Comment by LukeTowers** [comment]
@bennothommo can we add some sort of detection for this case to let users know that the failure state is likely caused by a slow or failing connection between their server and the package source?

**Comment by gusdgg** [comment]
Also, I think that maybe you can explicitly increase the max execution time for that particularly script with set_time_limit(600) or similar.
